src/components/edit-movie/edit-movie.tsx

```diff
@@ -2,30 +2,68 @@ import classNames from 'classnames';
 import { AppButton } from 'components/app-button/app-button';
 import { Genres } from 'components/menu/menu';
 import { Modal } from 'components/modal/modal';
+import { useState } from 'react';
+import { useAppDispatch } from 'redux/hooks';
+import { addMovieAction, updateMovieAction } from 'redux/movies';
+import { Movie } from 'types/movie';
 import styles from './edit-movie.module.scss';
 
 const genres = [Genres.Comedy, Genres.Crime, Genres.Documentary, Genres.Horror];
 
+type MovieFormValues = Partial<Movie>;
+
 interface Props {
   title: string;
+  movie?: Movie;
   onClose: () => void;
+  action: 'add' | 'edit';
 }
 
-export const EditMovie = ({ title, onClose }: Props) => {
+export const EditMovie = ({ title, movie, onClose, action }: Props) => {
+  const dispatch = useAppDispatch();
+
+  const [movieData, setMovieData] = useState<MovieFormValues | undefined>(
+    movie,
+  );
+
   const handleSubmit = () => {
+    if (!movieData) {
+      return;
+    }
+    if (!isValidMovie(movieData)) {
+      return;
+    }
```

Comment:
You may combine these into one expression:
```suggestion
    if (!movieData || !isValidMovie(movieData)) {
      return;
    }
```